src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -171,18 +170,18 @@ public Task ExecuteChildContentAsync()
         /// <returns>A <see cref="Task"/> that on completion returns the rendered child content.</returns>
         /// <remarks>
         /// Child content is only executed once. Successive calls to this method or successive executions of the 
-        /// returned <see cref="Task{string}"/> return a cached result.
+        /// returned <see cref="Task{TagHelperContent}"/> return a cached result.
         /// </remarks>
-        public async Task<string> GetChildContentAsync()
+        public async Task<TagHelperContent> GetChildContentAsync()
         {
             if (_childContent == null)
             {
-                _startWritingScope();
+                _startTagHelperWritingScope();
                 await _executeChildContentAsync();
-                _childContent = _endWritingScope().ToString();
+                _childContent = _endTagHelperWritingScope();
             }
 
-            return _childContent;
+            return new DefaultTagHelperContent().SetContent(_childContent);
```

Comment:
I'll concede the point for now since we're pushing to get this in. @sornaks file something in regards to `TagHelperExecutionContext` being smarter about new `TagHelperContent` allocations; as @dougbu we could get really nifty here, heck we could even be tricky with the type of `TagHelperContent` we give back to the user where we conditionally copy depending on if it gets modified.